src/mqtt/MQTT.cpp

```diff
@@ -142,6 +142,7 @@ void MQTT::onReceive(char *topic, byte *payload, size_t length)
                 if (strcmp(e.channel_id, channels.getGlobalId(ch.index)) == 0 && e.packet && ch.settings.downlink_enabled) {
                     LOG_INFO("Received MQTT topic %s, len=%u\n", topic, length);
                     meshtastic_MeshPacket *p = packetPool.allocCopy(*e.packet);
+                    p->via_mqtt = true; // Mark that the packet was received via MQTT
```

Comment:
do we also need to mark `via_mqtt` in the _if_ block above (for `mqtt.json_enabled`)?

Reply:
I thought this could be considered as a client <-> device connection. However, when I now look at it again, apparently it checks whether the "sender" is **not** equal to its own owner ID:
https://github.com/meshtastic/firmware/blob/f0b8430c6e3ffae33daca5375c0e037c5c3f5d8f/src/mqtt/MQTT.cpp#L99

Isn't this feature to allow people to inject a packet on their mesh via JSON? Now this looks very odd to me as it seems you could spoof a text message as coming from another node if you send this on an MQTT server.

Reply:
very odd indeed. I wonder if this `sender` spoofing is also a source of duplicates. but let's leave this alone for now.